tools/crm_resource.c

```diff
@@ -651,6 +651,13 @@ main(int argc, char **argv)
             rc = -ENXIO;
         }
     }
+    
+    if(require_resource && rsc_id == NULL)
+    {
+        CMD_ERR("Must supply a resource id with -r");
+        rc = -ENXIO;
+        goto bail;
+    }
```

Comment:
Let's do this check before we connect to the CIB. That way, if the user didn't specify a resource, we error out more quickly.

Reply:
Correct :) That's a much better way to . 
Thanks :)